integr8ly/walkthrough-applications
# INTLY-3042 UI changes to CRUD App

## Description
To improve the overall user experience of the solution web explorer, the UI of the CRUD app from walkthrough 1 should be updated

This includes:

* Centering the layout
* Moving over to the Patternfly framework
* Adding information component explaining the concept behind the legacy app
* Clearing fields after submit for snappy UI feel
* Any other design changes to improve look & feel of the app

## Verification
* With an openshift 4 cluster, install integreatly using operator source 
```
apiVersion: operators.coreos.com/v1
kind: OperatorSource
metadata:
  name: integreatly-operators
  namespace: openshift-marketplace
spec:
  authorizationToken: {}
  displayName: Integreatly Operators
  endpoint: 'https://quay.io/cnr'
  publisher: Integreatly Publisher
  registryNamespace: integreatly
  type: appregistry
```
* Create Installation CR that points to your own cluster following this example:
```
apiVersion: integreatly.org/v1alpha1
kind: Installation
metadata:
  name: example-installation
spec:
  type: workshop
  namespacePrefix: integreatly-
  routingSubdomain: apps.example.com
  masterUrl: http://console.apps.example.com
  selfSignedCerts: true
```
* Browse to solution web explorer after installation and go through walkthrough 1 to integrate the CRUD app with the node js messaging app
* Find the deployment config of the CRUD app 
	* should be named `crud` in a namespace like `opentlc-mgr-tuto-xxxx`
* Update the crud deployment config yaml container image to `quay.io/kevfan/fruit-crud-app:latest`
* Save deployment config change and wait for pod to be deployed.
* Refresh/Visit route to the CRUD application
* Verify the application, now uses the updated UI
* Verify fruit can still be created
* Verify fruit can still be deleted
* Verify there is now a footer with the following content:
```
Basic CRUD application that represents a legacy system.
```
* Verify that integration still works correctly via creating fruit via node js messaging app

## Screenshots
#### Mobile View
![Screenshot from 2019-08-30 13-51-04](https://user-images.githubusercontent.com/24636860/64034807-473ff080-cb47-11e9-92ac-ed73f74d29c6.png)
#### Desktop View
![Screenshot from 2019-08-30 16-55-56](https://user-images.githubusercontent.com/24636860/64034808-473ff080-cb47-11e9-9c34-7dab6d47e2f1.png)
![Screenshot from 2019-08-30 16-56-23](https://user-images.githubusercontent.com/24636860/64034809-47d88700-cb47-11e9-9a90-38667718b73d.png)
